mods/beds/functions.lua

```diff
@@ -81,6 +81,21 @@ local function lay_down(player, pos, bed_pos, state, skip)
 
 	-- lay down
 	else
+
+		-- Check if bed is occupied
+		for _, other_pos in pairs(beds.bed_position) do
+			if vector.distance(bed_pos, other_pos) < 0.1 then
```

Comment:
Does this matter near world edges? `beds.can_dig` does not compensate floating point errors.

Reply:
Huh? I didn't write it so I'm not sure how it works.

Reply:
Isn't it better to write in bed metadata that this is already taken and release it after waking up?

Reply:
Done

Reply:
The stored bed pos is a node position (each component of the pos vector needs 16 bits). Lua numbers (doubles) have a mantissa of more than 32 bit. I'm pretty sure that there will be no floating point errors.
You could even put the whole vector into one number with `minetest.hash_node_position`.